lib/Doctrine/DBAL/Schema/Table.php

```diff
@@ -57,17 +57,16 @@ class Table extends AbstractAsset
     protected $_schemaConfig = null;
 
     /**
-     * @param string                 $tableName
-     * @param Column[]               $columns
-     * @param Index[]                $indexes
-     * @param UniqueConstraint[]     $uniqueConstraints
-     * @param ForeignKeyConstraint[] $fkConstraints
-     * @param mixed[]                $options
+     * @param array<int|string, Column>               $columns
```

Comment:
@Ocramius as for the definitions like this, it's not clear for the caller what an `int` or `string` keys mean. How does the caller know which keys to use in order to pass the columns to the method, and what is the meaning of those keys? The method just iterates over the array and doesn't use the key at all.

As a rule of thumb, for a union type, you should be able to explain the difference between the types and when to use what. E.g. `function bindParam(int|string $param, mixed $value)` means that the parameter is either bound by its position (`int`) or name (`string`). In the case of `array<int|string, Column>`, `int` and `string` keys mean "nothing" and "nothing" respectively.

Reply:
I think these cases should probably be fixed so it is only `array<int|string, Column>`. I only documented the way it is today and I think it is that way today just by chance and inconsistency due to there not being any types before.

Reply:
I don't agree with `array<int, Column>` in this case either. If a caller currently passes `array<string, Column>`, this constraint will force them to use `array_values()` for no real reason (the original code works just fine) — same as it's being added in this pull request.

The bottom line is, I wouldn't like us to enforce the type of the variables (keys) which we don't use..

Reply:
I can see both sides and I could go either way. I would rather get this merged so I will change it back.

Reply:
I can accept the `array<>` approach as long as the patch doesn't contain unnecessary calls to `array_values()`. They do indicate the problem and I'd rather have it fixed than masked.

Reply:
Take a look at this commit https://github.com/doctrine/dbal/pull/3565/commits/125265bd436523fee013e38c7e57d29b3c275995. I backed out the ones where we don't use the key and the type causes the 
 need to use `array_values`.